sfaira/data/store/carts/multi.py

```diff
@@ -131,19 +130,19 @@ def move_to_memory(self):
             v.move_to_memory()
 
     @property
-    def ratios(self):
+    def iterator_frequencies(self):
         """
         Define relative drawing frequencies from iterators for intercalation.
 
         Note that these can be float and will be randomly rounded during intercalation.
         """
-        if self._ratios is None:
-            gen_lens = np.array([v.n_batches for v in self.carts.values()])  # eg. [10, 15, 20]
+        if self._iterator_frequencies is None:
+            iterator_lens = np.array([v.n_batches for v in self.carts.values()])  # eg. [10, 15, 20]
```

Comment:
can you do this based on cart.n_obs ?

Reply:
Yes I guess that s more accurate here because this batching is lost here already because of the split into cells, it does create slightly weird behaviour for batch size 0 where we deliver pre defined, variably size partitions of the data, but we can comment this and accept it for now, that s an edge case anyway.